packages/Webkul/Shop/src/Http/Controllers/API/CategoryController.php

```diff
@@ -45,11 +45,11 @@ public function index(): JsonResource
     /**
      * Get filterable attributes for category.
      */
-    public function getAttributes($categoryId): JsonResource
+    public function getAttributes(): JsonResource
     {
-        $category = $this->categoryRepository->findOrFail($categoryId);
+        $category = $this->categoryRepository->find(request('category_id'));
```

Comment:
Only hit database query when `category_id` is present.

~~~php
if (request('category_id')) { ... }
~~~